src/main/java/de/uni_kl/cs/discodnc/curves/Curve_ConstantPool.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * This file is part of the Disco Deterministic Network Calculator.
+ *
+ * Copyright (C) 2018+ The DiscoDNC contributors
+ *
+ * Distributed Computer Systems (DISCO) Lab
+ * University of Kaiserslautern, Germany
+ *
+ * http://discodnc.cs.uni-kl.de
+ *
+ *
+ * The Disco Deterministic Network Calculator (DiscoDNC) is free software;
+ * you can redistribute it and/or modify it under the terms of the 
+ * GNU Lesser General Public License as published by the Free Software Foundation; 
+ * either version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ */
+
+package de.uni_kl.cs.discodnc.curves;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import de.uni_kl.cs.discodnc.AlgDncBackend;
+import de.uni_kl.cs.discodnc.Calculator;
+import de.uni_kl.cs.discodnc.numbers.NumBackend;
+
+/**
+ * Constant pool for {@link Curve}. Stores constants based on {@link NumBackend} 
+ * and [{@link AlgDncBackend}.
+ * 
+ * @author matyesz
+ *
+ */
+public enum Curve_ConstantPool {
+
+    ZERO_CURVE {
+        @Override
+        protected Curve createCurve() {
+            return Curve.getFactory().createZeroCurve();
+        }
+    },
+    ZERO_SERVICE_CURVE {
+        @Override
+        protected Curve createCurve() {
+            return Curve.getFactory().createZeroService();
+        }
+    },
+    INFINITE_SERVICE_CURVE {
+        @Override
+        protected Curve createCurve() {
+            return Curve.getFactory().createZeroDelayInfiniteBurst();
+        }
+    },
+    ZERO_ARRIVAL_CURVE {
+        @Override
+        protected Curve createCurve() {
+            return Curve.getFactory().createZeroArrivals();
+        }
+    },
+    INFINITE_ARRIVAL_CURVE {
+        @Override
+        protected Curve createCurve() {
+            return Curve.getFactory().createInfiniteArrivals();
+        }
+    };
+    
+    private final Map<String, Curve> pool = new HashMap<>();
+
+    /**
+     * Returns the value of the constant from the pool.
+     * @return T the constant casted to type T
+     */
+    public <T extends Curve> T get() {
+        final String key = getKey();
+        Curve result = pool.get(key);
+        if (result == null) {
+            result = createCurve();
+            pool.put(key, result);
+         }
+        return (T) result;
```

Comment:
Can we resolve the unchecked cast here?

Reply:
I will add a @Suppresswarning("unchecked") as I do not see any other solution to resolve this issue without dirty hacking and complex codes. It is obvious that it will never fail.